### PR TITLE
Add optimized CuTeDSL KDA short convolution

### PR DESCRIPTION
**Pull request by drisspg**
### Human note
Speed

### Agent Note

- add a CuTeDSL fused causal depthwise convolution + SiLU implementation for the KDA short-convolution path
- add first-order input and weight gradients with FP32 accumulation and FP32 time-tile partial reduction
- add configurable/tunable launch schedules and compile-time-specialized convolution widths
- route the fused KDA training example through the optimized width-4 kernel for the batch-1, no-initial-convolution-state path
- add correctness, custom-op, full-graph compile, CUDA Graph, alignment, generic-width, and tuning tests

## GB300 results

Shape: `B=1, T=16384, C=12288, W=4`, BF16, NVIDIA GB300.

| Implementation | Forward | Backward |
| --- | ---: | ---: |
| PyTorch | ~2666 us | ~5463 us |
| FLA Triton | ~214 us | ~1376 us |
| CuTeDSL, integrated custom op | ~131 us | ~654 us |

This is approximately:

- 20x faster than PyTorch forward
- 8.4x faster than PyTorch backward

The isolated winning forward kernel is approximately 119.5 us. Focused NCU replay measured 5.78 TB/s, 72.9% DRAM throughput, 64 registers/thread, and 43.4% achieved occupancy.

## Implementation details

- Forward uses packed BF16 register staging, FP32 accumulation, and fused SiLU.
- Input gradient recomputes the preactivation and SiLU derivative.
- Weight gradient emits FP32 per-time-tile partials and reduces them in Torch.
- Width is inferred from `weight.shape[1]` and compile-time specialized.
- Threads, channels/thread, and times/block use the shared CuTeDSL compile-and-benchmark tuning flow.
- Generic widths use constexpr loops and TensorSSA reduction. Width 4 retains a compile-time-pruned expression-tree fast path: generic loop/reduction forms were correct but measured 215-230 us forward on the primary shape.
- A lean default custom-op schema avoids scalar configuration dispatch overhead in the production K3 path, while explicit configurations remain available.

## Validation

- `53 passed` across the new short-convolution tests and existing related KDA CuTeDSL suites
- forward, input gradient, and weight gradient checked for widths 1, 3, 4, and 5
- BF16 results checked against PyTorch and an FP64 measuring-stick reference
- `torch.library.opcheck`
- `torch.compile(fullgraph=True)` for default and explicit-config paths
- CUDA Graph capture/replay
- misaligned contiguous-storage handling
- default fused `B1/T16384/H32/D128` training step
- complete small model with `torch.compile(fullgraph=True, mode="reduce-overhead")`

## Test commands

```bash
prek run ruff --files \
  attn_gym/linear/kda/short_conv/cute.py \
  attn_gym/linear/kda/short_conv/__init__.py \
  test/test_kda_short_conv_cute.py \
  examples/kda_training.py
prek run ruff-format --files \
  attn_gym/linear/kda/short_conv/cute.py \
  attn_gym/linear/kda/short_conv/__init__.py \
  test/test_kda_short_conv_cute.py \
  examples/kda_training.py
gpu-run 0 -- bash -lc 'source ~/.venvs/nightly/bin/activate && PYTHONPATH=$PWD pytest -q test/test_kda_short_conv_cute.py test/test_kda_cute_forward.py test/test_kda_fused_gate_bwd.py'
gpu-run 0 -- bash -lc 'source ~/.venvs/nightly/bin/activate && PYTHONPATH=$PWD python examples/kda_training.py --backend fused --steps 1'
```

### CODE REVIEW

**Comment by drisspg** [comment]
see https://github.com/pytorch/pytorch/issues/192902 for some details, obvi the swiglu fusion is a lil special cased bu I mgiht native op this

File: attn_gym/linear/kda/short_conv/cute.py [hunk 1]
```diff
@@ -0,0 +1,1015 @@
...
+        A contiguous CUDA BF16 tensor with the same shape as ``x``.
+    """
+    _validate_inputs(x, weight)
+    if forward_config is None and input_grad_config is None and weight_grad_config is None:
+        for name, config in (
+            ("forward_config", CausalConv1dSiluForward.default_config),
+            ("input_grad_config", CausalConv1dSiluInputGradient.default_config),
+            ("weight_grad_config", CausalConv1dSiluWeightGradientPartials.default_config),
+        ):
+            _validate_config(config, x.shape[2], name)
```
[the remark this answers is not in the document]
Comment (drisspg) [reply attn_gym/linear/kda/short_conv/cute.py:983]: Fixed in follow-up PR #267. Defaults now select the gcd-compatible packed channel width, retain the lean width-4 path when valid, and add the fallback to tuning candidates. Added eager forward/backward and fullgraph coverage for C=5 and C=6.
